applications/FluidDynamicsApplication/custom_utilities/fluid_calculation_utilities.h

```diff
@@ -0,0 +1,171 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_FLUID_CALCULATION_UTILITIES_H)
+#define KRATOS_FLUID_CALCULATION_UTILITIES_H
+
+// system includes
+#include <tuple>
+#include <type_traits>
+
+// External includes
+
+// Project includes
+#include "geometries/geometry.h"
+#include "includes/node.h"
+#include "includes/ublas_interface.h"
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos classes
+///@{
+
+class KRATOS_API(FLUID_DYNAMICS_APPLICATION) FluidCalculationUtilities
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using NodeType = Node<3>;
+
+    using GeometryType = Geometry<NodeType>;
+
+    ///@}
+    ///@name Static Operations
+    ///@{
+
+    /**
+     * @brief Evaluates given list of variable pairs at gauss point locations at step
+     *
+     * Example:
+     *      double density;
+     *      array_1d<double, 3> velocity
+     *      EvaluateInPoint(rGeometry, rShapeFunction, Step,
+     *                      std::tie(density, DENSITY),
+     *                      std::tie(velocity, VELOCITY));
+     *
+     *      The above evaluation will fill density, and velocity variables with gauss point
+     *      evaluated DENSITY and VELOCITY at gauss point with shape function values rShapeFunction
+     *      in the geometry named rGeometry.
+     *
+     * @tparam TRefVariableValuePairArgs
+     * @param[in] rGeometry                 Geometry to get nodes
+     * @param[in] rShapeFunction            Shape function values at gauss point
+     * @param[in] Step                      Step to be evaluated
+     * @param[in/out] rValueVariablePairs   std::tuple<TDataType, const Variable<TDataType>> list of variables.
+     */
+    template <class... TRefVariableValuePairArgs>
+    static void EvaluateInPoint(
+        const GeometryType& rGeometry,
+        const Vector& rShapeFunction,
+        const int Step,
+        const TRefVariableValuePairArgs&... rValueVariablePairs)
```

Comment:
```suggestion
        const TRefVariableValuePairArgs&... rValueVariablePairs,
        const int Step = 0)
```
By doing this we avoid the method below.